### PR TITLE
Refactor packages

### PR DESCRIPTION
**Pull request by vivianjeng**
**Goal:** make changing config in unirep easier for unirep-social usage. Synchronize circuits, contracts, and genUserState
- link `unirep` package with `git+https://github.com/Unirep/Unirep.git#alpha-monorepo` and add it to dependencies
- change config in `unirep/config/index.ts` and then run `yarn build` to have the latest circuits and contracts
- use `@zk-kit/identity` to generate unirep identity
- use
  ```
  import { config, crypto, circuits, contracts, core } from 'unirep'
  ```
  to call functions in `@unirep/config`, `@unirep/crypto`, `@unirep/circuits`, `@unirep/contracts`, `@unirep/core`

### CODE REVIEW

**Comment by vuvth** [comment]
Can we remove `artifacts`? I think we overwrite it every time we build project.

File: package.json [hunk 1]
```diff
@@ -38,12 +39,9 @@
         "typechain": "^7.0.1"
     },
     "dependencies": {
+        "@openzeppelin/contracts": "^4.5.0",
         "@typechain/ethers-v5": "^9.0.0",
         "@typechain/hardhat": "^5.0.0",
-        "@unirep/circuits": "git+https://github.com/Unirep/circuits.git",
```
Comment (vimwitch) [inline package.json:43]: We should keep these dependencies but point them to a version number. Then the packages can be linked locally for development if needed.

**Comment by vivianjeng** [comment]
@vimwitch @vuvth ready for review.
It links to unirep github repository instead of `yarn link`.

**Comment by vivianjeng** [comment]
@vimwitch @vuvth ready for review.
It links to unirep github repository instead of `yarn link`.

**Comment by vimwitch** [comment]
The point of monorepos is to house many separate packages in a single repo. Linking them all into one defeats the purpose, it would be easier to simply have a single codebase without many packages.

What is the plan for publishing this? Will we use the `unirep` package name, or continue using `@unirep/crypto`, `@unirep/contracts`, etc?

**Comment by vimwitch** [comment]
If you want to install packages from a monorepo using a git url you can use [GitPkg](https://gitpkg.vercel.app/).

This would install `@unirep/circuits` from the main branch of the unirep repo:
`yarn add 'https://gitpkg.now.sh/Unirep/Unirep/packages/circuits?master'`

But IMO we should keep these individual packages up to date on npm and link them locally.

**Comment by vimwitch** [comment]
Another thing I noticed, having `unirep-social` depend on `unirep` is unclear to me. The backend now only depends on `unirep-social` and transitively gets `unirep`, which contains 4 other packages. It took me quite a while to figure it out.

What if the backend wants to use a different version of `@unirep/crypto`? Having a single package silently add many packages seems like an anti-pattern.

**Comment by vuvth** [comment]
> What is the plan for publishing this? Will we use the `unirep` package name, or continue using `@unirep/crypto`, `@unirep/contracts`, etc?

I think we will publish with `@unirep/crypto`, `@unirep/contracts`, etc.

**Comment by vuvth** [comment]
> What if the backend wants to use a different version of `@unirep/crypto`?

Before we publish packages, we will ensure any package have same version so I don't think developer want to use different version of unirep.

**Pull request closed without merging**